src/language-server/config/config.ts

```diff
@@ -79,7 +81,29 @@ const clientConfig = z.object({
 export type ClientConfigFormat = z.infer<typeof clientConfig>;
 
 const roverConfig = z.object({
-  bin: z.string().optional(),
+  bin: z
+    .preprocess(
+      (val) => val || which.sync("rover", { nothrow: true }) || undefined,
+      z.string({
+        message:
+          "Rover binary not found. Please either install it system-wide in PATH, or provide the `bin` option. Please also ensure that the binary is executable.",
+      }),
+    )
+    .refine(
+      (bin) => {
+        try {
+          // is executable?
+          accessSync(bin, fsConstants.X_OK);
+          // is a file and not a directory?
+          return statSync(bin).isFile();
+        } catch {
+          return false;
+        }
+      },
+      {
+        message: "Rover binary is not executable",
```

Comment:
```suggestion
        message: "Rover binary is installed but not not executable. Please ensure the rover binary is executable for...",
```

I'm not quite sure how to finish this sentence, but would it sense to provide a call to action here? At the very least, I'd consider mentioning that you were able to find the binary, but just couldn't execute it.

Reply:
```suggestion
        message: "Rover binary is not marked as an executable. If you are using OS X or Linux, ensure to set the executable bit.",
```